src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/EC/ECKeyPemTests.cs

```diff
@@ -0,0 +1,255 @@
+IDAeBgNVBAMMF1JTQSAzODQtYml0IENlcnRpZmljYXRlMEwwDQYJKoZIhvcNAQEB
+BQADOwAwOAIxANrMIthuZxV1Ay4x8gbc/BksZeLVEInlES0JbyiCr9tbeM22Vy/S
+9h2zkEciMuPZ9QIDAQABo1AwTjAdBgNVHQ4EFgQU5FG2Fmi86hJOCf4KnjaxOGWV
+dRUwHwYDVR0jBBgwFoAU5FG2Fmi86hJOCf4KnjaxOGWVdRUwDAYDVR0TBAUwAwEB
+/zANBgkqhkiG9w0BAQUFAAMxAEzDg/u8TlApCnE8qxhcbTXk2MbX+2n5PCn+MVrW
+wggvPj3b2WMXsVWiPr4S1Y/nBA==
+-----END CERTIFICATE-----
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgQHs5HRkpurXDPaabivT2IaRoyYt
+Isuk92Ner/JmgKjYoSumHVmSNfZ9nLTVjxeD08pD548KWrqmJAeZNsDDqQ==
+-----END PUBLIC KEY-----");
+            ECParameters ecParameters = ExportParameters(key, false);
+            ECParameters expected = EccTestData.GetNistP256ReferenceKey();
+            EccTestBase.ComparePublicKey(expected.Q, ecParameters.Q, isEqual: true);
+        }
+
+        [Fact]
+        public void ImportFromPem_Spki_PrecedingMalformedPem()
+        {
+            using TAlg key = CreateKey();
+            key.ImportFromPem(@"
+-----BEGIN CERTIFICATE-----
+$$ I AM NOT A PEM
+-----END CERTIFICATE-----
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgQHs5HRkpurXDPaabivT2IaRoyYt
+Isuk92Ner/JmgKjYoSumHVmSNfZ9nLTVjxeD08pD548KWrqmJAeZNsDDqQ==
+-----END PUBLIC KEY-----");
+            ECParameters ecParameters = ExportParameters(key, false);
+            ECParameters expected = EccTestData.GetNistP256ReferenceKey();
+            EccTestBase.ComparePublicKey(expected.Q, ecParameters.Q, isEqual: true);
+        }
+
+        [Fact]
+        public void ImportFromPem_Spki_AmbiguousKey_Spki()
+        {
+            string pem = @"
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgQHs5HRkpurXDPaabivT2IaRoyYt
+Isuk92Ner/JmgKjYoSumHVmSNfZ9nLTVjxeD08pD548KWrqmJAeZNsDDqQ==
+-----END PUBLIC KEY-----
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgQHs5HRkpurXDPaabivT2IaRoyYt
+Isuk92Ner/JmgKjYoSumHVmSNfZ9nLTVjxeD08pD548KWrqmJAeZNsDDqQ==
+-----END PUBLIC KEY-----";
+            using TAlg key = CreateKey();
+            AssertExtensions.Throws<ArgumentException>("input", () => key.ImportFromPem(pem));
+        }
+
+        [Fact]
+        public void ImportFromPem_Spki_AmbiguousKey_EncryptedPkcs8()
+        {
+            string pem = @"
+-----BEGIN ENCRYPTED PRIVATE KEY-----
+MIHgMEsGCSqGSIb3DQEFDTA+MCkGCSqGSIb3DQEFDDAcBAjVvm4KTLb0JgICCAAw
+DAYIKoZIhvcNAgkFADARBgUrDgMCBwQIuHgfok8Ytl0EgZDkDSJ9vt8UvSesdyV+
+Evt9yfvEjiP/6yITq59drw1Kcgp6buOCVCY7LZ06aD6WpogiqGDYMuzfvqg5hNFp
+opSAJ/pvHONL5kyAJLeNyG9c/mR2qyrP2L9gL0Z5fB9NyPejKTLi0PXMGQWdDTH8
+Qh0fqdrNovgFLubbJFMQN/MwwIAfIuf0Mn0WFYYeQiBJ3kg=
+-----END ENCRYPTED PRIVATE KEY-----
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEgQHs5HRkpurXDPaabivT2IaRoyYt
+Isuk92Ner/JmgKjYoSumHVmSNfZ9nLTVjxeD08pD548KWrqmJAeZNsDDqQ==
+-----END PUBLIC KEY-----";
+            using TAlg key = CreateKey();
+            AssertExtensions.Throws<ArgumentException>("input", () => key.ImportFromPem(pem));
```

Comment:
I don't see a test that says what happens to this input when sent to the encrypted PEM import.  (Either success, or still throwing). (Perhaps I'm just bad at searching/remembering)